docker-compose fail to build after PR#61

**Describe the bug**
docker-compose fail to build after PR #61

**To Reproduce**
checkout to 1462c800ee3341bd710d970a2c9fa8b5d5d862a2
```
docker-compose up --build
```


**OS:**
 - Windows Subsystem of Linux: Ubuntu-20.04



**Additional context**
error message:
```
[+] Building 8.4s (19/22)                                                                                docker:default
 => [twp internal] load build definition from Dockerfile                                                           0.0s
 => => transferring dockerfile: 505B                                                                               0.0s
 => [twp internal] load .dockerignore                                                                              0.0s
 => => transferring context: 76B                                                                                   0.0s
 => [twp internal] load metadata for docker.io/library/alpine:3.18                                                 1.0s
 => [twp internal] load metadata for docker.io/library/node:20-alpine3.18                                          1.0s
 => [twp internal] load metadata for docker.io/library/golang:1.21-alpine3.18                                      1.0s
 => [twp frontend-builder 1/6] FROM docker.io/library/node:20-alpine3.18@sha256:b1789b7be6aa16afd642eaaaccdeeeb33  0.0s
 => [twp runner 1/4] FROM docker.io/library/alpine:3.18@sha256:eece025e432126ce23f223450a0326fbebde39cdf496a85d8c  0.0s
 => [twp backend-builder 1/6] FROM docker.io/library/golang:1.21-alpine3.18@sha256:110b07af87238fbdc5f1df52b00927  0.0s
 => [twp internal] load build context                                                                              0.1s
 => => transferring context: 150.81kB                                                                              0.1s
 => CACHED [twp frontend-builder 2/6] WORKDIR /app                                                                 0.0s
 => CACHED [twp frontend-builder 3/6] COPY ./frontend/package.json ./frontend/package-lock.json ./                 0.0s
 => CACHED [twp frontend-builder 4/6] RUN npm ci                                                                   0.0s
 => CACHED [twp frontend-builder 5/6] COPY frontend .                                                              0.0s
 => ERROR [twp frontend-builder 6/6] RUN npm run build                                                             7.1s
 => CACHED [twp backend-builder 2/6] WORKDIR /app                                                                  0.0s
 => CACHED [twp backend-builder 3/6] COPY ./go.mod ./go.sum ./                                                     0.0s
 => CACHED [twp backend-builder 4/6] RUN go mod download                                                           0.0s
 => [twp backend-builder 5/6] COPY . .                                                                             0.4s
 => CANCELED [twp backend-builder 6/6] RUN go build -o twp                                                         6.9s
------
 > [twp frontend-builder 6/6] RUN npm run build:
0.655
0.655 > frontend@0.0.0 build
0.655 > tsc && vite build
0.655
6.798 src/pages/home/index.tsx(41,81): error TS2322: Type '{ id: number; name: string; imgUrl: string; isIndex: boolean; }' is not assignable to type 'IntrinsicAttributes & Props'.
6.798   Property 'isIndex' does not exist on type 'IntrinsicAttributes & Props'.
6.798 src/pages/home/index.tsx(55,81): error TS2322: Type '{ id: number; name: string; imgUrl: string; isIndex: boolean; }' is not assignable to type 'IntrinsicAttributes & Props'.
6.798   Property 'isIndex' does not exist on type 'IntrinsicAttributes & Props'.
------
failed to solve: process "/bin/sh -c npm run build" did not complete successfully: exit code: 2
```
_Originally mention by @x06lan in https://github.com/jykuo-love-shiritori/twp/issues/70#issuecomment-1832033106_

--- a/.github/workflows/frontend.yml
+++ b/.github/workflows/frontend.yml
@@ -4,11 +4,11 @@ on:
   workflow_dispatch:
   pull_request:
     branches:
-      - "main"
+      - 'main'
 
 jobs:
   lint:
-    runs-on: "ubuntu-latest"
+    runs-on: 'ubuntu-latest'
     steps:
       - name: Checkout Repository
         uses: actions/checkout@v3
@@ -21,3 +21,18 @@ jobs:
 
       - name: Run Linter
         run: cd frontend && npm run lint
+
+  build:
+    runs-on: 'ubuntu-latest'
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v3
+
+      - name: Setup Node
+        uses: actions/setup-node@v4
+
+      - name: Install Dependencies
+        run: cd frontend && npm ci
+
+      - name: Run Build
+        run: cd frontend && npm run build

--- a/frontend/src/pages/home/index.tsx
+++ b/frontend/src/pages/home/index.tsx
@@ -38,7 +38,7 @@ const Home = () => {
             if (data.id < 5) {
               return (
                 <Col xs={6} md={3} key={index}>
-                  <GoodsItem id={data.id} name={data.name} imgUrl={data.imgUrl} isIndex={true} />
+                  <GoodsItem id={data.id} name={data.name} imgUrl={data.imgUrl} />
                 </Col>
               );
             }
@@ -52,7 +52,7 @@ const Home = () => {
             if (data.id < 9 && data.id > 4) {
               return (
                 <Col xs={6} md={3} key={index}>
-                  <GoodsItem id={data.id} name={data.name} imgUrl={data.imgUrl} isIndex={true} />
+                  <GoodsItem id={data.id} name={data.name} imgUrl={data.imgUrl} />
                 </Col>
               );
             }

